src/System.Net.Sockets/src/System/Net/Sockets/_MultipleConnectAsync.cs

```diff
@@ -31,6 +39,10 @@ private enum State
 
         private object _lockObject = new object();
 
+        protected abstract bool RequiresUserConnectAttempt { get; }
```

Comment:
Are you suggesting that this entire class (the entire file, really) be duplicated per-platform? The code needs to accommodate an extra state in its state machine depending on whether or not the underlying platform supports multiple connection attempts on a single socket. The only way that I can think of to avoid the extra state is to try *every* connect on a new socket first on platforms that don't support multiple connection attempts, which adds significant overhead (for questionable benefit).